homeassistant/components/switchbot/light.py

```diff
@@ -77,16 +77,22 @@ async def async_turn_on(self, **kwargs: Any) -> None:
         ):
             kelvin = max(2700, min(6500, kwargs[ATTR_COLOR_TEMP_KELVIN]))
             await self._device.set_color_temp(brightness, kelvin)
-            return
+            self._attr_color_temp_kelvin = kelvin
```

Comment:
so, this is basically setting the states in HA before they are reported by switchbot, right? why do we need this? 

I think that deviates from how most integrations work. It also will make states flip if the new state is not applied, due to the "fake" state we set here.